src/BloomExe/Spreadsheet/SpreadsheetExporter.cs

```diff
@@ -75,7 +75,17 @@ public SpreadsheetExporter(ILanguageDisplayNameResolver langDisplayNameResolver)
 		{
 			var spreadsheet = ExportToFolder(dom, bookFolderPath, outputFolder, out string outputFilePath,
 				progress);
-			resultCallback(outputFilePath);
+			// ExportToFolder will return null if export fails, in which case we don't want to open/foreground the spreadsheet
+			if (spreadsheet != null)
+			{
+				resultCallback(outputFilePath);
+			}
+			else if (!progress.HaveProblemsBeenReported)
+			{
+				// Export failed but we have no progress problem messages. Make sure there is at least some error message
+				// so we aren't just failing silently
+				progress.MessageWithParams("Spreadsheet.ExportFailed", "", "Export failed: {0}", ProgressKind.Error, "");
```

Comment:
From a brief glance it looks possible for us to get here without any error messages having been sent to the progress dialog, so I just put a general export failure message that already exists to make sure there is something